pkg/metrics/providers/prometheus_test.go

```diff
@@ -170,6 +182,77 @@ func TestPrometheusProvider_RunQueryWithBasicAuth(t *testing.T) {
 
 }
 
+func TestPrometheusProvider_RunQueryWithBearerAuth(t *testing.T) {
+	t.Run("ok", func(t *testing.T) {
+		expected := `sum(envoy_cluster_upstream_rq)`
+		ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			promql := r.URL.Query()["query"][0]
+			assert.Equal(t, expected, promql)
+
+			header, ok := r.Header["Authorization"]
+			if assert.True(t, ok, "Authorization header not found") {
+				assert.True(t, strings.Contains(header[0], "Bearer"), "Bearer authorization header not found")
+			}
+
+			json := `{"status":"success","data":{"resultType":"vector","result":[{"metric":{},"value":[1545905245.458,"100"]}]}}`
+			w.Write([]byte(json))
+		}))
+		defer ts.Close()
+
+		clients := prometheusFake()
+
+		template, err := clients.flaggerClient.FlaggerV1beta1().MetricTemplates("default").Get(context.TODO(), "prometheus", metav1.GetOptions{})
+		require.NoError(t, err)
+		template.Spec.Provider.Address = ts.URL
+
+		secret, err := clients.kubeClient.CoreV1().Secrets("default").Get(context.TODO(), "prometheus-bearer", metav1.GetOptions{})
+		require.NoError(t, err)
+
+		prom, err := NewPrometheusProvider(template.Spec.Provider, secret.Data)
+		require.NoError(t, err)
+
+		val, err := prom.RunQuery(template.Spec.Query)
+		require.NoError(t, err)
+
+		assert.Equal(t, float64(100), val)
+	})
+
+	noResultTests := []struct {
+		name        string
+		queryResult string
+	}{
+		{name: "no values result", queryResult: `{"status":"success","data":{"resultType":"vector","result":[]}}`},
+		{name: "NaN result", queryResult: `{"status":"success","data":{"resultType":"vector","result":[{"metric":{},"value":[1643023250.379,"NaN"]}]}}`},
+	}
+
+	for _, tt := range noResultTests {
+		t.Run(tt.name, func(t *testing.T) {
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				json := tt.queryResult
+				w.Write([]byte(json))
+			}))
+			defer ts.Close()
+
+			clients := prometheusFake()
+
+			template, err := clients.flaggerClient.FlaggerV1beta1().
+				MetricTemplates("default").Get(context.TODO(), "prometheus", metav1.GetOptions{})
+			require.NoError(t, err)
+			template.Spec.Provider.Address = ts.URL
+
+			secret, err := clients.kubeClient.CoreV1().Secrets("default").Get(context.TODO(), "prometheus", metav1.GetOptions{})
+			require.NoError(t, err)
+
+			prom, err := NewPrometheusProvider(template.Spec.Provider, secret.Data)
+			require.NoError(t, err)
+
+			_, err = prom.RunQuery(template.Spec.Query)
+			require.True(t, errors.Is(err, ErrNoValuesFound))
+		})
+	}
+
```

Comment:
```suggestion

```
these test cases aren't required as they are exactly identical to the ones in `TestPrometheusProvider_RunQueryWithBasicAuth`